msg.go

```diff
@@ -43,6 +38,38 @@ const (
 	MsgSent      MsgStatus = "S"
 	MsgDelivered MsgStatus = "D"
 	MsgFailed    MsgStatus = "F"
+	NilMsgStatus MsgStatus = ""
+)
+
+// MsgDirection is the direction of a message
+type MsgDirection string
+
+// Possible values for MsgDirection
+const (
+	MsgIncoming     MsgDirection = "I"
+	MsgOutgoing     MsgDirection = "O"
+	NilMsgDirection MsgDirection = ""
+)
+
+// MsgPriority is the priority of our message
+type MsgPriority int
+
+// Possible values for MsgPriority
+const (
+	BulkPriority    MsgPriority = 100
+	DefaultPriority MsgPriority = 500
+	HighPriority    MsgPriority = 1000
+)
+
+// MsgVisibility is the visibility of a message
+type MsgVisibility string
+
+// Possible values for MsgVisibility
+const (
+	MsgVisible       MsgVisibility = "V"
+	MsgDeleted       MsgVisibility = "D"
+	MsgArchived      MsgVisibility = "A"
+	NilMsgVisibility MsgVisibility = ""
```

Comment:
Do we need a nil value - shouldn't new Msgs get the db default like for priority?